src/cargo/util/process_builder.rs

```diff
@@ -129,11 +131,11 @@ impl ProcessBuilder {
     }
 }
 
-pub fn process<T: AsRef<OsStr>>(cmd: T) -> CargoResult<ProcessBuilder> {
+pub fn process<T: AsRef<OsStr>, U: AsRef<OsStr>>(cmd: T, cwd: U) -> CargoResult<ProcessBuilder> {
```

Comment:
This may actually be best by just making the `cwd` field an `Option<OsString>` instead of making it required. `None` is equivalent to not passing this in and most of the time it shouldn't be required.